cmd/clusterctl/client/config.go

```diff
@@ -54,6 +55,45 @@ func (c *clusterctlClient) GetProviderComponents(provider string, providerType c
 	return components, nil
 }
 
+func (c *clusterctlClient) GenerateProvider(providerName string, providerType clusterctlv1.ProviderType, options ComponentsOptions) (Components, error) {
+
+	providerName, providerVersion, err := parseProviderName(providerName)
+	if err != nil {
+		return nil, err
+	}
+
+	// Gets the provider configuration (that includes the location of the provider repository)
+	providerConfig, err := c.configClient.Providers().Get(providerName, providerType)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get a client for the provider repository and read the provider components;
+	// during the process, provider components will be processed performing variable substitution, customization of target
+	// namespace etc.
+	// Currently we are not supporting custom yaml processors for the provider
+	// components. So we revert to using the default SimpleYamlProcessor.
+	repositoryClientFactory, err := c.repositoryClientFactory(RepositoryClientFactoryInput{Provider: providerConfig})
```

Comment:
```suggestion
	repositoryClient, err := c.repositoryClientFactory(RepositoryClientFactoryInput{Provider: providerConfig})
```

nit: the factory returns a client not another factory.